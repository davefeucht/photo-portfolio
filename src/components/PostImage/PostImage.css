.post-image {
  position: relative;
  grid-area: content;
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  transition: width var(--modal-transition-time), height var(--modal-transition-time)
}

.post-image img {
  transition: all var(--modal-transition-time), height var(--modal-transition-time)
}