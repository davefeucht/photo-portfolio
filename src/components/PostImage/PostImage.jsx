/****************
* PostImage component displays the image for a Post
****************/

import React, { useEffect } from "react";
import { observer } from 'mobx-react';
import './PostImage.css';

const PostImage = observer(({ stateStore }) => {
  return(
    <div className="post-image">
      <img src={stateStore.visiblePost.fullImageUrl}></img>
    </div>
  )
});

PostImage.displayName = 'PostImage';

export default PostImage;