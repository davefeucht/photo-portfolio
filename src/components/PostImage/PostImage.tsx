/** **************
* PostImage component displays the image for a Post
*************** */

import './PostImage.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { useState } from 'react';

import PostNavigationArrow from '../PostNavigationArrow/PostNavigationArrow';

interface PostImageProps {
    imageUrl: string,
    postHeight: number,
    previousPost: string,
    nextPost: string
}

const PostImage: React.FC<PostImageProps> = ({
    imageUrl,
    postHeight,
    previousPost,
    nextPost
}) => {
    const [arrowsVisible, setArrowsVisible] = useState<boolean>(false);

    const onMouseOverHandler = () => {
        setArrowsVisible(true);
    };

    const onMouseOutHandler = () => {
        setArrowsVisible(false);
    };

    return (
        <div className="post-image" onMouseOver={onMouseOverHandler.bind(this)} onMouseOut={onMouseOutHandler.bind(this)}>
            {arrowsVisible && <PostNavigationArrow postHeight={postHeight} direction="previous" postId={previousPost} />}
            <img src={imageUrl} />
            {arrowsVisible && <PostNavigationArrow postHeight={postHeight} direction="next" postId={nextPost} />}
        </div>
    );
};

PostImage.displayName = 'PostImage';

export default observer(PostImage);
