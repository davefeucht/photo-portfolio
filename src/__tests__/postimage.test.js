import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PostImage from '../components/PostImage/PostImage';
import stateStore from '../StateStore/store';

const previousPostId = 34;
const nextPostId = 36;

test('PostImage displays', () => {
  const store = new stateStore();
  let component;
  let tree;
  component = renderer.create(
      <MemoryRouter>
          <PostImage stateStore={store} previousPost={previousPostId} nextPost={nextPostId}></PostImage>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
