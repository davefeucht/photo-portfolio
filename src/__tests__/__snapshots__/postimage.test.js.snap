// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostImage displays 1`] = `
<div
  className="post-image"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="post-navigation-arrow active previous"
    style={
      {
        "left": "0px",
        "top": "0px",
      }
    }
  >
    <a
      href="/category/undefined/post/34"
      onClick={[Function]}
    >
      <div
        className="arrow"
      >
        <img
          src="./assets/images/chevron.svg"
        />
      </div>
    </a>
  </div>
  <img
    src={null}
  />
  <div
    className="post-navigation-arrow active next"
    style={
      {
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <a
      href="/category/undefined/post/36"
      onClick={[Function]}
    >
      <div
        className="arrow"
      >
        <img
          src="./assets/images/chevron.svg"
        />
      </div>
    </a>
  </div>
</div>
`;
